contracts/crowdsale/validation/IndividuallyCappedCrowdsale.sol

```diff
@@ -0,0 +1,50 @@
+pragma solidity ^ 0.4.18;
+
+import "../../math/SafeMath.sol";
+import "../Crowdsale.sol";
+import "../../ownership/Ownable.sol";
+
+/**
+ * @dev Crowdsale with per-user caps
+ */
+contract IndividuallyCappedCrowdsale is Crowdsale, Ownable {
+  using SafeMath for uint256;
+
+  mapping(address => uint256) public contributions;
+  mapping(address => uint256) public caps;
+
+  /**
+   * @param _beneficiary Address to be capped
+   * @param _cap Wei limit for individual contribution
+   */
+  function setUserCap(address _beneficiary, uint256 _cap) external onlyOwner {
+    caps[_beneficiary] = _cap;
+  }
+
+  /**
+   * @param _beneficiary Address whose cap is to be checked
+   * @return Current cap for individual user
+   */
+  function getUserCap(address _beneficiary) public view returns (uint256) {
+    return caps[_beneficiary];
+  }
+
+  /**
+   * @param _beneficiary Address of contributor
+   * @return User contribution so far
+   */
+  function getUserContribution(address _beneficiary) public view returns (uint256) {
+    return contributions[_beneficiary];
+  }
+
+  function _postValidatePurchase(address _beneficiary, uint256 _weiAmount) internal {
+    super._postValidatePurchase(_beneficiary, _weiAmount);
+    require(contributions[_beneficiary] <= caps[_beneficiary]);
+  }
+
+  function _updatePurchasingState(address _beneficiary, uint256 _weiAmount) internal {
```

Comment:
What's the purpose of this? 😋

Reply:
I think this is how the new crowdsale model flow works, we should add more documentation comments though

Reply:
We discussed adding this with @ajsantander, please see [above](https://github.com/OpenZeppelin/zeppelin-solidity/pull/744#discussion_r168559277) (link might work as intended as the comment is folded). The idea is to decouple validation from state tracking, taking into account future extensions.

Usage is now described in parent `Crowdsale`, should we repeat it here?